packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -392,22 +374,25 @@ class UpdateFSReport {
 class DevFS {
   /// Create a [DevFS] named [fsName] for the local files in [rootDirectory].
   DevFS(
-    VMService serviceProtocol,
+    vm_service.VmService serviceProtocol,
     this.fsName,
     this.rootDirectory, {
     @required OperatingSystemUtils osUtils,
+    @visibleForTesting bool disableUpload = false,
```

Comment:
I'm not 100% sure how well this works. I think this should be temporary until I can come up with a better refactor here: https://github.com/flutter/flutter/issues/55912

I like the named constructors because they seem more descriptive.